### PR TITLE
feat:  ComputeTask and ImageSource Support

### PR DESCRIPTION
**Pull request by roleyfoley**
## Intent of Change

- New feature (non-breaking change which adds functionality)
- Refactor

## Description

Adds support for configuring the AMI image of ec2 based components using
the image source configuration options.

Adds two AWS specific image source attributes to handle more explicit configuration options
- Source:AMI - Allows a user to explicitly define an AMI Id to use at the component level
- Source:SSMParam - Allows the user to define an SSM Parameter that holds the Image Id. The default value for these options is the AWS public parameters which list the latest recommended version of AWS provided AMI Images.

The default behaviour is to use the current AMI Ids defined in the
Region objects

Adds support and migrates all instance bootstrap configuration over to compute tasks to provide modular reusable configuration of instance based components. 

## Motivation and Context

This allows for users to have greater control over the AMI that is used for the virtual machine based components. Using the Public SSM parameters allows for deployments to automatically use the latest version of the image available from AWS at the time of deployment without having to depend on updates to hamlet to use the reference configuration 

The ability to use compute tasks allows for support of essentially any operating system that is used on the components as long as a compute task extension has been implemented for the required component compute tasks

## How Has This Been Tested?

Tested locally 

## Followup Actions

- [x] Requires: https://github.com/hamlet-io/engine/pull/1627 
- [x] Requires: https://github.com/hamlet-io/engine/pull/1625 
- [x] Requires: https://github.com/hamlet-io/engine/pull/1629
- [x] Requires: https://github.com/hamlet-io/engine/pull/1630
- [x] Requires: https://github.com/hamlet-io/engine/pull/1631 
- [x] Requires: #271

### CODE REVIEW

**Review by RossMurr4y — DISMISSED** [review]

**Comment by roleyfoley** [comment]
More to come moving back to draft

**Comment by ml019** [comment]
@roleyfoley #271 now merged - pls rebase and I'll review again

**Comment by roleyfoley** [comment]
@ml019 Rebased, fixed and should be good to go now

File: aws/attributesets/attributeset.ftl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[#ftl]
+
+[#assign AWS_OPERATINGSYSTEM_ATTRIBUTESET_TYPE = "aws_operatingsystem" ]
+[#assign AWS_COMPUTEIMAGE_ATTRIBUTESET_TYPE = "aws_computeimage" ]
+[#assign AWS_ECS_COMPUTEIMAGE_ATTRIBUTESET_TYPE = "aws_ecs_comptueimage"]
```
Comment (ml019) [inline aws/attributesets/attributeset.ftl:5]:
```suggestion
[#assign AWS_ECS_COMPUTEIMAGE_ATTRIBUTESET_TYPE = "aws_ecs_computeimage"]
```

**Review by ml019 — CHANGES_REQUESTED** [review]

**Review by ml019 — APPROVED** [review]

**Pull request merged**